engine/src/main/java/org/destinationsol/world/WorldBuilder.java

```diff
@@ -0,0 +1,119 @@
+    private ArrayList<FeatureGenerator> featureGenerators = new ArrayList<>();
+    private Context context;
+    private final int numberOfSystems;
+    private int systemsBuilt; //This field is for testing purposes
+
+    public WorldBuilder(Context context, int numSystems) {
+        this.context = context;
+        numberOfSystems = numSystems;
+        populateSolSystemGeneratorList();
+        populateFeatureGeneratorList();
+        systemsBuilt = 0;
+    }
+
+    /**
+     * This method uses reflection to retrieve all SolSystemGenerator classes. They are added to the list
+     * of SolSystemGenerators.
+     */
+    private void populateSolSystemGeneratorList() {
+
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(SolSystemGenerator.class)) {
+            try {
+                SolSystemGenerator solSystemGenerator = (SolSystemGenerator) generator.newInstance();
+                solSystemGenerators.add(solSystemGenerator);
+            } catch (InstantiationException | IllegalAccessException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * This method uses reflection to retrieve all concrete FeatureGenerator classes. They are added to the list
+     * of FeatureGenerators.
+     */
+    private void populateFeatureGeneratorList() {
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(FeatureGenerator.class)) {
+            if (!Modifier.isAbstract(generator.getModifiers())) {
+                try {
+                    FeatureGenerator featureGenerator = (FeatureGenerator) generator.newInstance();
+                    featureGenerators.add(featureGenerator);
+                } catch (InstantiationException | IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * This method initializes the SolSystemGenerators. How many generators are initialized depends on the number
+     * of SolSystems the world is set to have. When there are multiple types of SolSystemGenerators available, this
+     * method chooses randomly from the list of all generators to decide which to create.
+     *
+     * This method calls setFeatureGenerators() on each SolSystemGenerator to give each SolSystem access to other
+     * FeatureGenerators (Planets, Mazes, etc.) so they can be initialized and incorporated into the SolSystem.
+     */
+    private void initializeSolSystemGenerators() {
+        for (int i = 0; i < numberOfSystems; i++) {
+            SolSystemGenerator solSystemGenerator = solSystemGenerators.get(SolRandom.seededRandomInt(solSystemGenerators.size()));
+            solSystemGenerator.setFeatureGenerators(featureGenerators);
+            solSystemGenerator.build();
+            systemsBuilt++;
+        }
+    }
+
+    public void build() {
+        initializeSolSystemGenerators();
+    }
+
+    public int getSystemsBuilt() {
```

Comment:
What does this do? if this is only for testing, a comment to make that clear could be useful :)

Reply:
@IsaacLic @Cervator mentioned that putting comments on getters and setters is probably not necessary. The field it retrieves already has a comment by it.